packages/react-query-devtools/src/devtools.tsx

```diff
@@ -438,26 +438,23 @@ export const ReactQueryDevtoolsPanel = React.forwardRef<
 
   const queries = React.useMemo(() => {
     const unsortedQueries = queryCache.getAll()
-    const sorted = queriesCount > 0 ? [...unsortedQueries].sort(sortFn) : []
 
-    if (sortDesc) {
-      sorted.reverse()
+    if (queriesCount === 0) {
+      return []
     }
 
-    if (!filter) {
-      return sorted
-    }
-
-    let ranked = sorted.map(
-      (item) => [item, rankItem(item.queryHash, filter)] as const,
-    )
-
-    ranked = ranked.filter((d) => d[1].passed)
+    const filtered = filter
+      ? unsortedQueries.filter(
+          (item) => rankItem(item.queryHash, filter).passed,
+        )
+      : [...unsortedQueries]
 
-    ranked = ranked.sort((a, b) => compareItems(a[1], b[1]))
+    const sorted = sortFn
+      ? filtered.sort((a, b) => sortFn(a, b) * (baseSort as number))
```

Comment:
all of our usages of `useLocalStorage` provide the optional default value, so if we make it mandatory, we don't have to do type assertions here. Also, with the current implementation, there is a render cycle where the value is indeed `undefined`. We can fix that by passing the `defaultValue` directly to `useState` inside `useLocalStorage` and can then also get rid of this whole block:

https://github.com/TanStack/query/blob/357ec041a6fcc4a550f3df02c12ecc7bcdefbc05/packages/react-query-devtools/src/useLocalStorage.ts#L24-L27

Reply:
@TkDodo Not sure why but when i touch anything about `useLocalStorage` to get rid of initial `undefined` render for values, tests start to break.
Basically this code block stops devtools from being rendered:
```ts
// Do not render on the server
if (!isMounted()) return null
```
🤷

Reply:
ok, 🤷